src/sentry/static/sentry/app/views/discover/index.tsx

```diff
@@ -204,11 +201,29 @@ class DiscoverContainer extends React.Component<Props, State> {
     });
   };
 
-  renderNoAccess() {
+  renderNoAccess = () => {
+    const {router, organization} = this.props;
+
+    const redirectDiscoverQuery = () => (
+      <Redirect
+        router={router}
+        to={`/organizations/${organization.slug}/discover/results/`}
+      />
+    );
+
     return (
-      <AlertAsAny type="warning">{t("You don't have access to this feature")}</AlertAsAny>
+      <Feature
+        features={['discover-query']}
+        organization={organization}
+        renderDisabled={redirectDiscoverQuery}
```

Comment:
Ah I didn't anticipate how awkward this might be.

The features happen to be available in the `organization` prop. For example: https://github.com/getsentry/sentry/blob/2d94296414a0ffb5fd1634766595199695985f3b/src/sentry/static/sentry/app/views/eventsV2/resultsChart.tsx#L140

This should simplify into an if-else condition.